connector/saml/saml.go

```diff
@@ -334,19 +353,97 @@ func (p *provider) HandlePOST(s connector.Scopes, samlResponse string) (ident co
 	return ident, nil
 }
 
+func (p *provider) validateStatus(resp *response) error {
+	status := resp.Status
+	if status == nil {
+		return fmt.Errorf("response did not contain a Status")
+	}
+	statusCode := status.StatusCode
+	if statusCode == nil {
+		return fmt.Errorf("response did not contain a StatusCode")
+	}
+	if statusCode.Value != "urn:oasis:names:tc:SAML:2.0:status:Success" {
+		return fmt.Errorf("status code of the response was not Success")
+	}
+	return nil
+}
+
+func (p *provider) validateSubjectConfirmation(subject *subject) error {
```

Comment:
The relevant part in the spec [Profiles for the OASIS Security
Assertion Markup Language](https://www.oasis-open.org/committees/download.php/35389/sstc-saml-profiles-errata-2.0-wd-06-diff.pdf) is section 3.3 Bearer.

> URI: urn:oasis:names:tc:SAML:2.0:cm:bearer
> The subject of the assertion is [E47]the bearer ofconsidered to be an acceptable attesting entity for the
> assertion by the asserting party, subject to optional constraints on confirmation using the attributes that
> MAY be present in the <SubjectConfirmationData> element, as defined by [SAMLCore].
> If the intended bearer is known by the asserting party to be an entity other than the subject, then the
> asserting party SHOULD identify that entity to the relying party by including a SAML identifier representing
> it in the enclosing <SubjectConfirmation> element.
> If multiple attesting entities are to be permitted to use the assertion based on bearer semantics, then
> multiple <SubjectConfirmation> elements SHOULD be included.
> Example: The bearer of the assertion can confirm itself as the subject, provided the assertion is delivered
> in a message sent to "https://www.serviceprovider.com/saml/consumer" before 1:37 PM GMT on March
> 19th, 2004, in response to a request with ID "_1234567890".
```xml
<SubjectConfirmation Method="urn:oasis:names:tc:SAML:2.0:cm:bearer">
  <SubjectConfirmationData InResponseTo="_1234567890" Recipient="https://www.serviceprovider.com/saml/consumer" NotOnOrAfter="2004-03-19T13:27:00Z"
  </SubjectConfirmationData>
</SubjectConfirmation>
```

This is inline with the python implementation https://github.com/onelogin/python-saml/blob/master/src/onelogin/saml2/response.py#L234-L263. 

Multiple subject SubjectConfirmation can be in the assertion and at least one SubjectConfirmation must be valid. I will change this accordingly. Currently the request id is not available if I remember correctly.